sys/bloom/bloom.c

```diff
@@ -35,7 +35,7 @@ bloom_t *bloom_new(size_t size, size_t num_hashes, ...)
     }
 
     /* Allocate Bloom array */
-    if (!(bloom->a = calloc(ROUND(size), sizeof(char)))) {
+    if (!(bloom->a = calloc(sizeof(uint8_t), ROUND(size)))) {
```

Comment:
If we find such an esoteric platform then I am sure we will have other things to struggle with. The bloom filter operates on `uint8_t` only, so having `char` in the first place was also not correct